src/include/matrix.h

```diff
@@ -200,25 +200,19 @@ class Matrix {
     void pow(const size_t &exponent, Matrix<O> &output) const;
     inline void pow(const size_t &exponent);
 
-    /* Transpose (*this), and store the result in output
-     * The function whose parameters do not include output will change (*this)
-     * NOTE: output must have the same shape with the current matrix after transposition
+    /* Return the transposed matrix of (*this)
      *       If O and ELEMENT_TYPE are not same, all the elements will first be cast to
```

Comment:
This `If O and ELEMTN_TYPE...` should be removed, there is no `O` and `ELEMENT_TYPE` at all.